app/models/pipeline_run.rb

```diff
@@ -137,19 +137,20 @@ def update_job_status
     if all_stages_succeeded
       self.finalized = 1
       self.job_status = STATUS_CHECKED
+      save
+      notify_users if notify?
+      return
```

Comment:
Let's not have returns in the middle of functions.    I refactored this code earlier to get rid of them, because they prove error prone over time, particularly in environments with exceptions and hooks.

Reply:
ok